pymemcache/serde.py

```diff
@@ -16,12 +16,13 @@
 import logging
 from io import BytesIO
 import pickle
+import zlib
 
 FLAG_BYTES = 0
 FLAG_PICKLE = 1 << 0
 FLAG_INTEGER = 1 << 1
 FLAG_LONG = 1 << 2
-FLAG_COMPRESSED = 1 << 3  # unused, to main compatibility with python-memcached
```

Comment:
Any insight into how this flag  still maps back to python-memcached. Also would this flag cause any issues since two folks can use the same flag but different compression algorithms?



Also I wonder what compatibility we still have with python-memcached before this change and now.

Reply:
Relatedly, a way to handle a generally compressed value with different compressors is to introduce a prefix.

```py
serde = CompressedSerde(..., prefix="z")
```

... and then prepend that prefix on serialization (e.g. `value = prefix + "|" + value`). On deserialization, validate and strip the prefix.

Reply:
python-memcached used zlib by default but I don't think at this point there is any chance of compatibility since we are supporting multiple versions of pickle and we wouldn't be able to guarantee versions across the serialization. 

I like the idea of adding a prefix on there so we identify which compressor was used.   In the end the client is only going to have one (via serde) so its not going to be able to handle decompressing/compressing multiple versions but we could at least raise an error that says "Found keys compressed with a different serializer".

I don't know if we do this today though.   For example, we don't track pickle versions so we can select the correct one

Reply:
@jparise @jogo How strongly are you feeling that you want to be able to identify the compression algorithm between clients that are configured with different `serde`?     I have a feeling its not that high of a priority since we don't do that today with the pickle serializer but wanted to check

Reply:
I think to do it properly we'd have to do what we discussed before, which I'm not really a fan of:

https://github.com/pinterest/pymemcache/issues/53#issuecomment-124928276

Reply:
I agree it's not a high priority now. I mainly wanted to explicitly make sure we weren't trying to keep any python-memcached compatibility.